lib/include/public/CommonFields.h

```diff
@@ -24,6 +24,15 @@
 #define COMMONFIELDS_DEVICE_CLASS                            "DeviceInfo.Class"
 #define COMMONFIELDS_DEVICE_ORGID                            "DeviceInfo.OrgId"
 
+/* Ref. https://www.w3.org/TR/trace-context/#traceparent-header-field-values
+ - ParentId  - parent of the current span received via remote context propagation.
+ - SpanId    - current SpanId sent in 'traceparent' to remote service.
+ */
+#define COMMONFIELDS_DT_TRACEID                              "env_dt_traceId"
+#define COMMONFIELDS_DT_SPANID                               "env_dt_spanId"
+#define COMMONFIELDS_DT_TRACEFLAGS                           "env_dt_traceFlags"
+#define COMMONFIELDS_DT_PARENTID                             "parentId"
```

Comment:
My understanding is that `parentId` is meant to be used in events with Span semantics. Here we are adding it to every single event instead, which looks a bit weird. Are we doing this because supporting explicit Span semantics is more complicated? Would it cause problems if we decide to support explicit Span semantics later?

Reply:
There are two separate logical destinations here:

- first three: **traceId**, **spanId**, **traceFlags** - is our current span, what we would've sent in `traceparent` header.

- last one: **parentId** - is what we receive as our parent, what initiated us to run, what span is our parent. Developer focusing on client telemetry would receive this not as part of `traceparent` sent to the remote end, but as part of telemetry that their own code is emitting. That way we stitch together the caller (our parent) and our call into tree view.. Without this information attached on event - the Trace View explorer UX is not going to stitch the two operations on one time ladder chart.

So `spanid` -- is our current span id, `parentId` -- is our parent remote span id.

Reply:
>  Here we are adding it to every single event instead

Not every single event. Only for events on specific component logger, that are relevant to specific service interaction. However, if a client controls the initial seed of `traceId`, then the same `traceId` could be set to different loggers - `serviceA-REST-Logger`, `serviceB-REST-Logger`, `serviceC-REST-Logger`. Since they have share the same `traceId`, they all show up in one call tree.

The second example in `APITest` shows how you can selectively append the context directly on just one event, without using `SetContext`.. so both ways are possible, context - all events for specific logger, and local - just one event.

Reply:
Discussed more offline. Having a `parentId` for an event is a valid model even if the system doesn't follow the span model (see also [comment in TraceContext standard](https://github.com/w3c/trace-context/blob/main/http_header_format_rationale.md#spanparent-nomenclature)). Supporting explicit Span semantics is sort of overlapping with what OpenTelemetry does and out of scope for now.